galley/pkg/server/controller.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"time"
+
+	"istio.io/istio/pkg/probe"
+)
+
+const (
+	//DefaultProbeCheckInterval defines default probeCheck interval
+	DefaultProbeCheckInterval = 2 * time.Second
+	//DefaultLivenessProbeFilePath defines default livenessProbe filePath
+	DefaultLivenessProbeFilePath = "/healthLiveness"
+	//DefaultReadinessProbeFilePath defines default readinessProbe filePath
+	DefaultReadinessProbeFilePath = "/healthReadiness"
+)
+
+var (
+	livenessProbeController, readinessProbeController probe.Controller
+	galleyProbe                                       *probe.Probe
+)
+
+//StartProbeCheck start probe check for Galley
+func StartProbeCheck(livenessProbeOptions *probe.Options, readinessProbeOptions *probe.Options, stop <-chan struct{}) {
+	defer func() {
+		galleyProbe.SetAvailable(nil)
```

Comment:
A `nil` value indicates that the probe is ready/healthy. We need to set a non-nil value when the health check is stopped, e.g. `probe.SetAvailable(errors.New("not ready"))`

Reply:
done: validationProbe.SetAvailable(errors.New("stopped"))